JASP-Engine/JASP/R/descriptives.R

```diff
@@ -1445,36 +1445,41 @@ Descriptives <- function(jaspResults, dataset, options) {
 
-  data1= na.omit(as.data.frame(dataset))
+  data1 <- na.omit(dataset)
   means <- rowMeans(data1)
   subgroups <- 1:length(means)
-  data2 <- data.frame(subgroups = subgroups, means = means)
-  sixsigma <- qcc::qcc(data1, type = 'xbar', plot = F)
+  data2 <- data.frame(subgroups=subgroups, means=means)
+  sixsigma <- qcc::qcc(data1, type='xbar', plot=FALSE)
   center <- sixsigma$center
   sd1 <- sixsigma$std.dev
   UCL <- max(sixsigma$limits)
   LCL <- min(sixsigma$limits)
-
-  p <- ggplot2::ggplot(data2, ggplot2::aes(x = subgroups, y = means, group = 1))+ ggplot2::geom_line()+
-    ggplot2::ylab("Subgroup mean")+
-    ggplot2::xlab('Subgroup')+
-    ggplot2::theme_classic()+
-    ggplot2::scale_x_continuous(breaks = c(subgroups), limits = c(1,length(subgroups) + 1.5))+
-    ggplot2::geom_point(size = 2)+
-    ggplot2::geom_hline(yintercept = center, color = 'black')+
-    ggplot2::geom_hline(yintercept = c(center - sd1,center + sd1), linetype = "dashed", color = "red")+
-    ggplot2::geom_hline(yintercept = c(UCL,LCL), color = "red")+
-    ggplot2::geom_label(ggplot2::aes(length(subgroups)+1.2,center,label = sprintf("Mean = %g",round(center,3))))+
-    ggplot2::geom_label(ggplot2::aes(length(subgroups)+1.2,UCL,label = sprintf("UCL = %g",round(UCL,3))))+
-    ggplot2::geom_label(ggplot2::aes(length(subgroups)+1.2,LCL,label = sprintf("LCL = %g",round(LCL,3))))+
-    ggplot2::theme(axis.text.x = ggplot2::element_text(size= ggplot2::rel(1.4)), axis.text.y= ggplot2::element_text(size= ggplot2::rel(1.4)))+
-    ggplot2::theme(axis.text.x = ggplot2::element_text(size= ggplot2::rel(1.4)), axis.text.y= ggplot2::element_text(size= ggplot2::rel(1.4)))+
-    ggplot2::scale_y_continuous(limits = c(LCL,UCL), breaks = NULL)
+  dfLabel <- data.frame(
+    x <- length(subgroups) + 1.2,
+    y <- c(center, UCL, LCL),
+    l <- c(
+      gettextf("Mean = %g", round(center, 3)),
+      gettextf("UCL = %g",   round(UCL, 3)),
+      gettextf("LCL = %g",   round(LCL, 3))))
+
+  p <- ggplot2::ggplot(data2, ggplot2::aes(x = subgroups, y = means)) +
+    ggplot2::geom_line() +
+    ggplot2::geom_point(size = 2) +
+    ggplot2::geom_hline(yintercept =  center, color = 'black') +
+    ggplot2::geom_hline(yintercept=c(center - sd1, center + sd1), linetype = "dashed", color = "red") +
+    ggplot2::geom_hline(yintercept= c(UCL, LCL), color = "red") +
+    ggplot2::geom_label(data = dfLabel, mapping = ggplot2::aes(x = x, y = y, label = l),inherit.aes = FALSE) +
+    ggplot2::theme(axis.text.x = ggplot2::element_text(size= ggplot2::rel(1.4)), axis.text.y = ggplot2::element_text(size = ggplot2::rel(1.4))) +
+    ggplot2::theme(axis.text.x = ggplot2::element_text(size= ggplot2::rel(1.4)), axis.text.y = ggplot2::element_text(size = ggplot2::rel(1.4))) +
+    ggplot2::scale_y_continuous(name = "Subgroup mean" ,limits = c(LCL,UCL), breaks = NULL) +
+    ggplot2::scale_x_continuous(name = 'Subgroup', breaks = subgroups, limits = c(1, length(subgroups) + 1.5)) +
```

Comment:
If you look at the plot you see:
![image](https://user-images.githubusercontent.com/21319932/96117634-afd46300-0eea-11eb-8dc7-b1deafc2f247.png)

The x-axis doesn't look ideal. Base R solves this by only showing some of the labels:

![image](https://user-images.githubusercontent.com/21319932/96117753-d98d8a00-0eea-11eb-84f8-ce42d4810fa4.png)


In `ggplot2::scale_x_continuous(name = 'Subgroup', breaks = subgroups, limits = c(1, length(subgroups) + 1.5))` you can add the argument `labels` where you specify what text should be shown at each axis tick. So you could make this a vector with only `""` and then fill each 5th (or 10th, or 100th, depending on the data) tick with a number.